airsonic-main/src/main/java/org/airsonic/player/service/upnp/ApacheUpnpServiceConfiguration.java

```diff
@@ -0,0 +1,59 @@
+/*
+ This file is part of Airsonic.
+
+ Airsonic is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Airsonic is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Airsonic.  If not, see <http://www.gnu.org/licenses/>.
+
+ Copyright 2016 (C) Airsonic Authors
+ Based upon Subsonic, Copyright 2013 (C) Sindre Mehus
+ */
+package org.airsonic.player.service.upnp;
+
+import org.fourthline.cling.DefaultUpnpServiceConfiguration;
+import org.fourthline.cling.transport.impl.apache.StreamClientConfigurationImpl;
+import org.fourthline.cling.transport.impl.apache.StreamClientImpl;
+import org.fourthline.cling.transport.impl.apache.StreamServerConfigurationImpl;
+import org.fourthline.cling.transport.impl.apache.StreamServerImpl;
+import org.fourthline.cling.transport.spi.NetworkAddressFactory;
+import org.fourthline.cling.transport.spi.StreamClient;
+import org.fourthline.cling.transport.spi.StreamServer;
+
+import java.util.concurrent.Executors;
+
+/**
+ * UPnP configuration which uses Apache HttpComponents.  Needed to make UPnP work
+ * when deploying on Tomcat.
+ *
+ * @author Sindre Mehus
+ * @version $Id$
+ */
+public class ApacheUpnpServiceConfiguration extends DefaultUpnpServiceConfiguration {
+
+    public ApacheUpnpServiceConfiguration() {
+        super();
+    }
+
+    public ApacheUpnpServiceConfiguration(int streamListenPort) {
+        super(streamListenPort);
+    }
+
+    @Override
+    public StreamClient createStreamClient() {
+        return new StreamClientImpl(new StreamClientConfigurationImpl(Executors.newCachedThreadPool()));
+    }
```

Comment:
This is again more or less the same as the parent. The only difference is that the parent provides its own custom executor and we provide a generic one.

Reply:
Nevermind, the packages are different